### PR TITLE
🌱 Add more linters

### PR DESCRIPTION
**Pull request by elfosardo**
- gocritic
- gosec
- whitespace

### CODE REVIEW

File: test/suite_test.go [hunk 1]
```diff
@@ -214,7 +214,7 @@ func VerifyIronic(ironic *metal3api.Ironic) {
 	var nodeUUIDs []string
 	for idx := 0; idx < 100; idx++ {
 		node, err := nodes.Create(withTimeout, cli, nodes.CreateOpts{
-			Driver: drivers[rand.Intn(len(drivers))],
+			Driver: drivers[rand.Intn(len(drivers))], //nolint:gosec
```
Comment (tuminoid) [inline test/suite_test.go:217]:
```suggestion
			Driver: drivers[rand.Intn(len(drivers))], //nolint:gosec  // weak crypto is ok in tests
```

**Review by tuminoid — COMMENTED** [review]
Suggesting adding some comments

File: .golangci.yaml [hunk 2]
```diff
@@ -55,6 +55,8 @@ linters-settings:
     severity: medium
     confidence: medium
     concurrency: 8
+    excludes:
+      - G115
```
[the remark this answers is not in the document]
Comment (elfosardo) [reply .golangci.yaml:59]: I will add a note, just wanted to point out that I read contradictory opinions on G115, mostly negative ones, like the one you linked
we can only hope about the next version

[the remark this answers is not in the document]
Comment (tuminoid) [reply .golangci.yaml:60]: Comments are indeed negative, but as the article describes 1.62 is better and latest version is 1.63.x already. We can remove the exclude when bumping and see how bad it is, and decide to keep or remove.

I'm not arguing against adding the exclude right now, just asking for TODO note.

**Review by tuminoid — COMMENTED** [review]
/approve

**Pull request merged**